utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -114,143 +151,212 @@ object BroadcastUtils {
     errorUdf
   }
 
-  private def getMappingLambdaParam1(mappingTable: Broadcast[LocalMappingTable], defaultValueOpt: Option[Any]): AnyRef = {
-    defaultValueOpt match {
-      case None =>
-        param1: Any => {
-          val mt = mappingTable.value.map
-          mt.getOrElse(Seq(param1), null)
-        }
-      case Some(defaultValue) =>
-        param1: Any => {
-          val mt = mappingTable.value.map
-          mt.getOrElse(Seq(param1), defaultValue)
+  private def mapMultipleOutputs(mappingTable: Broadcast[LocalMappingTable], row: Option[Any], defaults: Map[String, Any]): Row = {
+    //TODO default values need extra validation
+    row match {
+      case Some(row: Row) => {
+        val rowValues: Map[String, Any] = row.getValuesMap[Any](mappingTable.value.outputColumns.keys.toSeq)
+        val values = rowValues.map{ case (field, value) => {
+          if (value == null) defaults(field)
+          else value
+        }}
+        new GenericRowWithSchema(values.toArray, row.schema)
+      }
+      case Some(_) => null
+      case None => {
+        if (defaults.isEmpty) null
+        else {
+          val values = mappingTable.value.outputColumns.values.toSeq.map(field => defaults.getOrElse(field, null))
+          Row.fromSeq(values)
         }
+      }
     }
   }
 
-  private def getMappingLambdaParam2(mappingTable: Broadcast[LocalMappingTable], defaultValueOpt: Option[Any]): AnyRef = {
-    defaultValueOpt match {
-      case None =>
-        (param1: Any, param2: Any) => {
-          val mt = mappingTable.value.map
-          mt.getOrElse(Seq(param1, param2), null)
+  private def getMappingLambdaParam1(mappingTable: Broadcast[LocalMappingTable], defaultValueMap: Map[String, Any], hasSingleOutput: Boolean = false): AnyRef = {
+    if (hasSingleOutput) {
+        defaultValueMap.get(mappingTable.value.outputColumns.values.head) match {
+          case None =>
+            param1: Any => {
+              val mt = mappingTable.value.map
+              mt.getOrElse(Seq(param1), null)
+            }
+          case Some(defaultValue) =>
+            param1: Any => {
+              val mt = mappingTable.value.map
+              mt.getOrElse(Seq(param1), defaultValue)
+            }
         }
```

Comment:
for example:
```suggestion
        val defaultValue = defaultValueMap.get(mappingTable.value.outputColumns.values.head).orNull
        (param1: Any => {
          val mt = mappingTable.value.map
          mt.getOrElse(Seq(param1), defaultValue)
        })
```

This seems to be common to all 1-10 param methods.